pkg/operator/apiserver/controller/auditpolicy/auditpolicy_controller.go

```diff
@@ -0,0 +1,136 @@
+package auditpolicy
+
+import (
+	"context"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/apiserver/audit"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/yaml"
+)
+
+type AuditPolicyController struct {
+	apiserverConfigLister                configv1listers.APIServerLister
+	kubeClient                           kubernetes.Interface
+	operatorClient                       v1helpers.OperatorClient
+	targetNamespace, targetConfigMapName string
+}
+
+// NewAuditPolicyController create a controller that watches the config.openshift.io/v1 APIServer object
+// and reconciles a ConfigMap in the target namespace with the audit.k8s.io/v1 policy.yaml file.
+func NewAuditPolicyController(
+	targetNamespace string,
+	targetConfigMapName string,
+	apiserverConfigLister configv1listers.APIServerLister,
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	configInformers configinformers.SharedInformerFactory,
+	kubeInformersForTargetNamesace kubeinformers.SharedInformerFactory,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &AuditPolicyController{
+		operatorClient:        operatorClient,
+		apiserverConfigLister: apiserverConfigLister,
+		kubeClient:            kubeClient,
+		targetNamespace:       targetNamespace,
+		targetConfigMapName:   targetConfigMapName,
+	}
+
+	return factory.New().WithSync(c.sync).ResyncEvery(10*time.Second).WithInformers(
+		configInformers.Config().V1().APIServers().Informer(),
+		kubeInformersForTargetNamesace.Core().V1().ConfigMaps().Informer(),
+	).ToController("AuditPolicyController", eventRecorder.WithComponentSuffix("audit-policy-controller"))
+}
+
+func (c *AuditPolicyController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	operatorConfigSpec, _, _, err := c.operatorClient.GetOperatorState()
```

Comment:
`operatorClient` is backed by a lister i assume?

Reply:
yes, it is: dynamicStaticPodOperatorClient